goalc/compiler/compilation/ConstantPropagation.cpp

```diff
@@ -245,19 +247,43 @@ Compiler::ConstPropResult Compiler::constant_propagation_dispatch(const goos::Ob
 }
 
 s64 Compiler::get_constant_integer_or_error(const goos::Object& in, Env* env) {
-  auto prop = try_constant_propagation(in, env);
+  const auto prop = try_constant_propagation(in, env);
   if (prop.value.is_pair()) {
-    auto head = prop.value.as_pair()->car;
+    const auto& head = prop.value.as_pair()->car;
     if (head.is_symbol()) {
-      auto head_sym = head.as_symbol();
-      auto enum_type = m_ts.try_enum_lookup(head_sym.name_ptr);
+      const auto& head_sym = head.as_symbol();
+      const auto enum_type = m_ts.try_enum_lookup(head_sym.name_ptr);
       if (enum_type) {
+        if (m_settings.check_for_requires) {
+          // Check if the enum has been required or not
+          if (enum_type->m_metadata.definition_info.has_value() &&
+              !env->file_env()->m_missing_required_files.contains(
+                  enum_type->m_metadata.definition_info->filename) &&
+              env->file_env()->m_required_files.find(
+                  enum_type->m_metadata.definition_info->filename) ==
+                  env->file_env()->m_required_files.end() &&
+              !str_util::ends_with(enum_type->m_metadata.definition_info->filename,
+                                   env->file_env()->name() + ".gc")) {
+            lg::warn("Missing require in {} for {} over {}", env->file_env()->name(),
+                     enum_type->m_metadata.definition_info->filename, head_sym.name_ptr);
+            env->file_env()->m_missing_required_files.insert(
+                enum_type->m_metadata.definition_info->filename);
+          }
+        }
         bool success;
-        u64 as_enum =
-            enum_lookup(prop.value, enum_type, prop.value.as_pair()->cdr, false, &success);
+        // TODO - changed false to true so you'd actually once again get a useful error (something
+        // wasn't found in the enum instead of just "cant be used as a constant") Ramifications of
+        // this?
```

Comment:
I think you're right, and this was probably just a copy-paste mistake. If this enum lookup fails, you are doomed.